internal/templates/templates_testutils.go

```diff
@@ -35,15 +37,18 @@ func (s *TemplateTestSuite) Validate(
 	cases []TestCase,
 ) {
 	for _, c := range cases {
-		checkFunc, err := s.Template.Instantiate(c.Param)
-		if c.ExpectInstantiationError {
-			s.Error(err, "param should have caused error but did not raise one")
-			continue
-		}
-		for _, obj := range ctx.Objects() {
-			diagnostics := checkFunc(ctx, obj)
-			s.compareDiagnostics(c.Diagnostics, diagnostics)
-		}
+		s.Run(fmt.Sprintf("%+v", c.Param), func() {
+			checkFunc, err := s.Template.Instantiate(c.Param)
+			if c.ExpectInstantiationError {
```

Comment:
should this be `c.ExpectInstantiationError && err == nil`?

Reply:
`s.Error(err)` inside basically fails if `err != nil` so it's correct.